haystack/document_stores/in_memory/document_store.py

```diff
@@ -50,15 +51,200 @@ def __init__(
                                               To choose the most appropriate function, look for information about your embedding model.
         """
         self.storage: Dict[str, Document] = {}
-        self._bm25_tokenization_regex = bm25_tokenization_regex
+        self.bm25_tokenization_regex = bm25_tokenization_regex
         self.tokenizer = re.compile(bm25_tokenization_regex).findall
-        algorithm_class = getattr(rank_bm25, bm25_algorithm)
-        if algorithm_class is None:
-            raise ValueError(f"BM25 algorithm '{bm25_algorithm}' not found.")
-        self.bm25_algorithm = algorithm_class
+
+        self.bm25_algorithm = bm25_algorithm
+        self.bm25_algorithm_inst = self._dispatch_bm25()
         self.bm25_parameters = bm25_parameters or {}
         self.embedding_similarity_function = embedding_similarity_function
 
+        # Global BM25 statistics
+        self._avg_doc_len: float = 0.0
+        self._freq_doc: Counter = Counter()
```

Comment:
I think it `self._vocab_freq` would make more sense as name, since this holds the frequency of each token

Reply:
TBH I think the naming of this variable is very tricky because it is used to hold "how many document s in the store contain a particular token" which is supplied later to compute inverse document frequency; on the contrary, I think `_vocab_freq` express the semantics more like "how many times a token appears in the entire corpus". I am more inclined to keep the document frequency name but add more comments here to clarify the intent of this variable. But I am open to how others think :D

Reply:
Updated in [this commit](https://github.com/deepset-ai/haystack/pull/7549/commits/687bef3b51145400058c746734217cf5b59b132d)